packages/bot-skeleton/src/scratch/hooks/contextmenu.js

```diff
@@ -177,3 +177,36 @@ Blockly.ContextMenu.blockCommentOption = function (block) {
 
     return comment_option;
 };
+
+/**
+ * Make a context menu option for detaching the current block.
+ * deriv-bot: Use Blockly's implementation.
+ * @param {!Blockly.BlockSvg} block The block where the right-click originated.
+ * @return {!Object} A menu option, containing text, enabled, and a callback.
+ * @package
+ */
+Blockly.ContextMenu.blockDetachOption = function (block) {
+    const canDisconnect = block
+        .getConnections_()
+        .find(connection => connection.isConnected() && !connection.targetConnection?.sourceBlock_.isShadow_);
```

Comment:
Better change it for `some` method, since we need a boolean for `canDisconnect` variable.
```suggestion
        .some(connection => connection.isConnected() && !connection.targetConnection?.sourceBlock_.isShadow_);
```

Reply:
good point stranger

Reply:
changed 👍